app/src/main/java/com/eveningoutpost/dexdrip/utils/BgToSpeech.java

```diff
@@ -0,0 +1,139 @@
+package com.eveningoutpost.dexdrip.utils;
+
+import android.content.Context;
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.preference.PreferenceManager;
+import android.speech.tts.TextToSpeech;
+import android.util.Log;
+
+import com.eveningoutpost.dexdrip.UtilityModels.Constants;
+
+import java.text.DecimalFormat;
+import java.util.Locale;
+
+/**
+ * Created by adrian on 07/09/15.
+ */
+public class BgToSpeech {
+
+    private static BgToSpeech instance;
+    private final Context context;
+
+    private TextToSpeech tts = null;
+
+    private long timestamp = 0;
+
+    public static BgToSpeech getSingleton(Context context){
+
+        if(instance == null ||  instance.context != context) {
+            instance = new BgToSpeech(context);
+        }
+        return instance;
+    }
+
+    private BgToSpeech(Context context){
+        this.context = context;
+    }
+
+    public void speak(final double value, long timestamp){
+        final SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(context);
+
+        if (! prefs.getBoolean("bg_to_speech", false)){
+            return;
+        }
+
+        if(this.timestamp == timestamp){
```

Comment:
ah is this how you guarded against it being called multiple times in a row if multiple things kicked off the service?